src/Coordination/KeeperContext.cpp

```diff
@@ -0,0 +1,37 @@
+#include <Coordination/KeeperContext.h>
+#include <Coordination/KeeperConstants.h>
+#include <Core/SettingsEnums.h>
+#include <Common/ErrorCodes.h>
+#include <Common/logger_useful.h>
+#include <Core/SettingsFields.h>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int BAD_ARGUMENTS;
+}
+
+DECLARE_SETTING_ENUM(KeeperApiVersion);
+IMPLEMENT_SETTING_ENUM(KeeperApiVersion, ErrorCodes::BAD_ARGUMENTS,
+    {{"ZOOKEEPER_COMPATIBLE", KeeperApiVersion::ZOOKEEPER_COMPATIBLE},
+     {"WITH_FILTERED_LIST", KeeperApiVersion::WITH_FILTERED_LIST},
+     {"WITH_MULTI_READ", KeeperApiVersion::WITH_MULTI_READ},
+     {"WITH_CHECK_NOT_EXISTS", KeeperApiVersion::WITH_CHECK_NOT_EXISTS}});
+
+void KeeperContext::initialize(const Poco::Util::AbstractConfiguration & config)
```

Comment:
Larger refactor of `KeeperContext` is in https://github.com/ClickHouse/ClickHouse/pull/50098 so I didn't modify it much in this PR